medallion/backends/memory_backend.py

```diff
@@ -133,17 +132,14 @@ def get_objects(self, api_root, collection_id, filter_args, allowed_filters):
             for collection in collections:
                 if "id" in collection and collection_id == collection["id"]:
 
-                    if filter_args:
-                        full_filter = BasicFilter(filter_args)
-                        objs.extend(
-                            full_filter.process_filter(
-                                collection.get("objects", []),
-                                allowed_filters,
-                                collection.get("manifest", []),
-                            ),
-                        )
-                    else:
-                        objs.extend(collection.get("objects", []))
+                    full_filter = BasicFilter(filter_args)
+                    objs.extend(
+                        full_filter.process_filter(
+                            collection.get("objects", []),
+                            allowed_filters,
+                            collection.get("manifest", []),
+                        ),
+                    )
```

Comment:
objs.extend() is not required as we only extracting objects from a single collection.

```suggestion
                    objs = full_filter.process_filter(
                        collection.get("objects", []),
                        allowed_filters,
                        collection.get("manifest", []),
                    )
                    break
```